blockstore.go

```diff
@@ -219,12 +219,7 @@ func (bs *blockstore) AllKeysChan(ctx context.Context) (<-chan cid.Cid, error) {
 				return
 			}
 
-			kb, err := dshelp.BinaryFromDsKey(ds.RawKey(e.Key))
-			if err != nil {
-				log.Warningf("error parsing binary from DsKey: %s", err)
-				continue
-			}
-			k, err := cid.Cast(kb)
+			k, err := dshelp.DsKeyToCid(ds.RawKey(e.Key))
```

Comment:
In the previous version there was this comment: https://github.com/ipfs/go-ipfs-blockstore/pull/89/files#diff-4d0eb65b9891ffe23bd03c7596f2b573a5ecd5378e939ab7801acb5dca97b147L222
```
			// need to convert to key.Key using key.KeyFromDsKey.
			k, err := dshelp.DsKeyToCid(ds.RawKey(e.Key))
```